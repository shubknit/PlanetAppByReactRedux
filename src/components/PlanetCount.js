import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { getFilteredData } from '../selectors/filteredData';

export const PlanetCount = ({ planetData }) => (
    <div className ="page-header">
		<div className = 'content-container'>
			<h1 className = "page-header__title"> Viewing { planetData.length } Planets </h1>
		</div>
	</div>
)

const mapStateToProps = (state) => {
    return {
        planetData: getFilteredData(state.planetList, state.filters)
    }
};

export default connect(mapStateToProps)(PlanetCount);