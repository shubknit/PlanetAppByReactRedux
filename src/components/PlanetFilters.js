import React from 'react';
import { PureComponent } from 'react';
import { connect } from 'react-redux';
import { SetTextFilter } from '../actions/filters';

class PlanetFilters extends PureComponent {

	constructor(props){
		super();
		this.filteredData = this.filteredData.bind(this);
	}

    filteredData(event) {
         this.props.onTextChange(event.target.value)
    }

    render() {
        return (
            <div className="content-container">
		<div className="input-group">
			<div className="input-group__item">
				<input type="text" className="text-input" placeholder="Search planets"  onChange = {this.filteredData}/>
			</div>
		</div>
	</div>
        )
    }

}


const mapDispatchToProps = (dispatch) => {
    return {
        onTextChange: (text) => dispatch(SetTextFilter(text))
    }
}

export default connect(undefined, mapDispatchToProps)(PlanetFilters);