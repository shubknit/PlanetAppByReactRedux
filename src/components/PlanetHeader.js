import React from 'react';
import { Link } from 'react-router-dom';
import HomeIcon from 'react-icons/lib/fa/home';
import { connect } from 'react-redux';
import { endUserLogin } from '../actions/login';

export const PlanetHeader = (props) => {
	const logoutHandler = () => {
		props.logoutUser();
	}

    return (
        <header className ="header">
		<div className ="content-container">
			<div className = 'header__content'>
				<Link to= '/user' className = 'header__title'>
					<HomeIcon/>
				</Link>
				<h1 className ='header__title'>Planet App </h1>
				{props.isAuthenticated ?
					<Link to= '/login' className = 'button button--link' onClick = {logoutHandler}>
					 Logout
					</Link> : ''
				}	
			</div>	
			
			 
		</div>
		</header>
    )
}
const mapStateToProps  = (state) => {
    return {
        isAuthenticated: state.user.isAuthenticated
    }
}

const mapDispatchToProps = (dispatch) => {
	return {
		logoutUser : () => dispatch(endUserLogin())
	}
}

export default connect(mapStateToProps, mapDispatchToProps)(PlanetHeader);