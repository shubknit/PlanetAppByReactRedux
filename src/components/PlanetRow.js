import React from 'react';
import { Link } from 'react-router-dom';

export const PlanetRow = (props) => {
	const { planetRowData } = props,
	 linkStyle = {
		fontSize: 30 - props.order * 2
	}
	return(
			<Link to = {`planet-detail/${planetRowData.id}`} className = 'list-item'
		    style = {linkStyle}>		
				<div className = "col-1">
					{ planetRowData.name } 
			    </div>	
				<div className = "list">
					{ planetRowData.population }
				</div>
			</Link>	
	)
}	