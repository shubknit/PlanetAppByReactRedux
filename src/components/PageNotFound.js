import React from 'react';
import { PlanetHeader } from './PlanetHeader';
export const PageNotFound = () => (
	<div className ="404">
		<PlanetHeader/>
		<div className = 'content-container'>
				<h1 className ='page-header__title'>PageNotFound </h1> 
		</div>	
	</div>
)