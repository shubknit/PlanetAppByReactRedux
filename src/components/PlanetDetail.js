import React from 'react';
import PlanetHeader  from './PlanetHeader';
import { connect }  from 'react-redux';

export const PlanetDetail = (props) => {

	const id = props.match.params.id,
	 [planetData] = props.planetList.filter(item => item.id == id);
	
	return(
		<div className ="heading">
			<PlanetHeader/>
			<div className = 'page-header'>
				<div className = 'content-container'>	
					<h1 className ='page-header__title'>Planet {planetData.name} Detail  </h1> 
				</div>
			</div>
			<div className = 'content-container'>
				<div className = 'list-header'>
					<div> Rotation Period</div>
					<div> Orbital Period</div>
					<div> Diameter</div>
					<div> Population</div>
				</div>
				<div className = 'list-body'>
					<div className = 'list-item'>
						<div> {planetData.rotation_period} </div>
						<div> {planetData.orbital_period} </div>
						<div> {planetData.diameter} </div>
						<div> {planetData.population} </div>
					</div>
				</div>	
			</div>	
		</div>
	)
}

const mapStateToProps = (state) => {
	return {
		planetList: state.planetList
	}
}

export default connect(mapStateToProps)(PlanetDetail);