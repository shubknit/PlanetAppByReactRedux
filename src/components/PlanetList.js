import React from 'react';
import { PlanetRow } from './PlanetRow';
import { getFilteredData } from '../selectors/filteredData';
import { connect } from 'react-redux';

export const PlanetList = ({ planetData }) => {
    return (

        <div className = "content-container">
				<div className = 'list-header'>
					<div> Name</div>
					<div> Population</div>
				</div>
				<div className = 'list-body'>
					{	planetData.length ? 
						planetData.map((data,i) => 
						<PlanetRow key = {i} planetRowData = {data} order = {i}/> )
						: <div className="list-item list-item--message"><span>No planets</span></div>	
					}	
				</div>
			</div>


    )
}

const mapStateToProps = (state) => {
    return {
        planetData: getFilteredData(state.planetList, state.filters)
    }
};

export default connect(mapStateToProps)(PlanetList);