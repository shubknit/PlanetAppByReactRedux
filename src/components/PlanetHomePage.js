import React from 'react';
import { Component } from 'react';
import PlanetHeader  from './PlanetHeader';
import PlanetFilters  from './PlanetFilters';
import  PlanetList  from './PlanetList';
import  PlanetCount from './PlanetCount';
import { Redirect } from 'react-router';

export class PlanetHomePage extends Component {
    constructor() {
        super();
    }
    render() {
        return (
            <div className = 'home-page'>
				<div>
                    <PlanetHeader/> 
                    <PlanetCount/>
                    <div className="content-container">
                        <PlanetFilters/>
                        <PlanetList/>
                    </div>
                    </div>

			</div>
        )

    }
}

