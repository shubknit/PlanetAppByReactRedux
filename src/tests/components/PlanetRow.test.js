import React from 'react';
import { PlanetRow } from '../../components/PlanetRow';
import { shallow } from 'enzyme';
import { PlanetListData } from '../planetListData';

test('should render PlanetRow correctly', () => {
	const wrapper = shallow(<PlanetRow planetRowData = {PlanetListData[0]}/>);
	expect(wrapper).toMatchSnapshot();
});