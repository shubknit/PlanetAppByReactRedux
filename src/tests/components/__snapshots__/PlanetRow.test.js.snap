// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render PlanetRow correctly 1`] = `
<Link
  className="list-item"
  replace={false}
  style={
    Object {
      "fontSize": NaN,
    }
  }
  to="planet-detail/1"
>
  <div
    className="col-1"
  >
    mars
  </div>
  <div
    className="list"
  >
    100
  </div>
</Link>
`;
